src/object_accessor.hpp

```diff
@@ -64,14 +64,14 @@ ValueType Object::get_property_value(ContextType& ctx, StringData prop_name)
 
 template <typename ValueType, typename ContextType>
 void Object::set_property_value_impl(ContextType& ctx, const Property &property,
-                                     ValueType value, bool try_update, bool is_default)
+                                     ValueType value, bool try_update, bool update_only_diff, bool is_default)
 {
     ctx.will_change(*this, property);
 
     auto& table = *m_row.get_table();
     size_t col = property.table_column;
     size_t row = m_row.get_index();
-    if (is_nullable(property.type) && ctx.is_null(value)) {
+    if (is_nullable(property.type) && ctx.is_null(value) && (!update_only_diff || !table.is_null(col, row))) {
```

Comment:
The `update_only_diff` check needs to be pushed into a separate `if` inside the body of this or it'll continue on to the rest of the function.

Reply:
Indeed. Thanks